apps/service-portal/src/hooks/useModules/useModules.ts

```diff
@@ -6,72 +6,37 @@ import { useFeatureFlagClient } from '@island.is/react/feature-flags'
 
 import { useStore } from '../../store/stateProvider'
 import { ActionType } from '../../store/actions'
-import {
-  featureFlaggedModules,
-  featureFlaggedCompanyModules,
-  ModuleKeys,
-  companyModules,
-} from '../../store/modules'
+import { ModuleKeys, companyModules } from '../../store/modules'
 
 export const useModules = () => {
   const featureFlagClient = useFeatureFlagClient()
   const [{ modules }, dispatch] = useStore()
   const { userInfo } = useAuth()
 
-  async function filterModulesBasedOnFeatureFlags() {
-    const flagValues = await Promise.all(
-      featureFlaggedModules.map((moduleKey) => {
-        const capKey = moduleKey.charAt(0).toUpperCase() + moduleKey.slice(1)
-        return featureFlagClient.getValue(
-          `isServicePortal${capKey}ModuleEnabled`,
-          false,
+  async function filterEnabledModules() {
+    const isCompany = userInfo?.profile?.subjectType === 'legalEntity'
+    const moduleEntries = Object.entries(modules) as [
+      ModuleKeys,
+      ServicePortalModule,
+    ][]
+    const filteredModules: Record<string, ServicePortalModule> = {}
+
+    for (const [moduleKey, module] of moduleEntries) {
+      let enabled = true
+
+      if (enabled && isCompany && !companyModules.includes(moduleKey)) {
+        enabled = false
```

Comment:
redundant enabled check in the statement above, and might be better to `continue` here to skip `enabled` checks everywhere else.

Reply:
Just a thought regarding if reduce improves readability? :
```typescript
const filteredModules = moduleEntries.reduce((acc, [moduleKey, module]) => {
  const isMissingFromCompanyModules = isCompany && !companyModules.includes(moduleKey)
  const isFeatureFlagged = !isMissingFromCompanyModules && module.featureFlag && Boolean(await featureFlagClient.getValue(module.featureFlag, false))
  if (isFeatureFlagged) {
    return {
      ...acc,
      [moduleKey]: module,
    }
  }
  return acc
}, {})
```

Reply:
In general I like verbose variable names to help with readability but some of the other suggestions I disagree with:

- `continue` _could_ make sense here, but I was conditioned to avoid the keyword years ago, since it can make code harder to read or follow, similar to old `goto` and multiple returns. 🤓
- Regarding `reduce`, I agree strongly with [Jake Archibald](https://twitter.com/jaffathecake/status/1213077702300852224) on this one. I would especially like to have an eslint rule banning reduce-based object/array construction because its bad performance (`O(n^2)`) is ... at best, unnecessarily slow ... at worst, causing dangerous CPU-bottlenecks on single-threaded JS runtime running on slow mobile devices or overloaded server processes. 😬

I kept the redundant `enabled` check for consistency, idea being that a module is enabled unless any condition makes it not enabled. Sure I could implement this somehow fancier but I like the KISS principle.

Also, I must admit I stared at your suggestion for over 5 minutes to try and understand if it was correct, only to finally figure out that it's not...

<details>
<summary>Feel free to compare the implementations on your own, or expand this...</summary>

It does not return modules for normal users or company modules for companies unless the modules also have a configured featureFlag which is true.

Here's my stab to fix it in the GitHub UI. I always ended up with a ton of double negations, which are really confusing. Still not 100% sure if the logic is correct or not. 😆

```
const filteredModules = moduleEntries.reduce((acc, [moduleKey, module]) => {
  const isMissingFromCompanyModules = isCompany && !companyModules.includes(moduleKey)
  const isMissingFeatureFlag = !isMissingFromCompanyModules && module.featureFlag && !Boolean(await featureFlagClient.getValue(module.featureFlag, false))
  if (!isMissingFromCompanyModules && !isMissingFeatureFlag) {
    return {
      ...acc,
      [moduleKey]: module,
    }
  }
  return acc
}, {})
```

</details>

Thanks for taking the time, but I think I'll keep this as is. ❤️